nixos/modules/services/web-apps/mastodon.nix

```
@@ -28,7 +28,9 @@ let
 
     TRUSTED_PROXY_IP = cfg.trustedProxy;
   }
-  // (if cfg.smtp.authenticate then { SMTP_LOGIN  = cfg.smtp.user; } else {})
+  // (if cfg.redis.enableUnixSocket then { REDIS_URL = "unix:///run/redis/redis.sock"; } else {})
```

```suggestion
  // (if cfg.redis.enableUnixSocket then { REDIS_URL = "unix://${config.services.redis.unixSocket}"; } else {})
```
with an additional assertion that `config.services.redis.unixSocket != null`